tests/test_apiproxy.py

```diff
@@ -182,13 +209,51 @@ def test_reply_source(self):
         with open(os.path.join(dirname, "encrypted_msg.asc")) as fobj:
             data = fobj.read()
 
-        self.assertTrue(self.api.reply_source(s, data))
+        reply = self.api.reply_source(s, data)
+        assert isinstance(reply, Reply)
+        assert reply.uuid
+        assert reply.filename
+
+    @dastollervey_datasaver
+    def test_reply_source_with_uuid(self):
+        s = self.api.get_sources()[0]
+        dirname = os.path.dirname(__file__)
+        with open(os.path.join(dirname, "encrypted_msg.asc")) as fobj:
+            data = fobj.read()
+
+        msg_uuid = "e467868c-1fbb-4b5e-bca2-87944ea83855"
+        reply = self.api.reply_source(s, data, msg_uuid)
+        assert reply.uuid == msg_uuid
+
+    @dastollervey_datasaver
+    def test_download_submission(self):
+        s = self.api.get_all_submissions()[0]
+
+        self.assertFalse(s.is_read)
+
+        # We need a temporary directory to download
+        tmpdir = tempfile.mkdtemp()
+        _, filepath = self.api.download_submission(s, tmpdir)
+
+        # Uncomment the followig part only on QubesOS
```

Comment:
not necessary for merge, just a thought: It would be cool to do something like [this](6be674f00fe1b7939e75450f6b4bca2b023be548) not to skip the whole test but to run the commented-out logic when we're on qubes (using a distinguishing attribute for Qubes in [platform](https://docs.python.org/3/library/platform.html) stdlib)

Reply:
This one we should skip as we run the tests normally (with test data) even on Qubes many times.

Reply:
valid point, we're not necessarily using the proxy, resolving